src/reanimated2/UpdateProps.ts

```diff
@@ -43,27 +43,46 @@ if (shouldBeUseWeb()) {
     }
   };
 } else {
-  updatePropsByPlatform = (
-    viewDescriptors: SharedValue<Descriptor[]>,
-    updates: StyleProps | AnimatedStyle,
-    _: ViewRefSet<any> | undefined
-  ): void => {
-    'worklet';
+  if (global._IS_FABRIC) {
+    updatePropsByPlatform = (
+      viewDescriptors: SharedValue<Descriptor[]>,
+      updates: StyleProps | AnimatedStyle,
+      _: ViewRefSet<any> | undefined
+    ): void => {
+      'worklet';
 
-    for (const key in updates) {
-      if (ColorProperties.indexOf(key) !== -1) {
-        updates[key] = processColor(updates[key]);
+      for (const key in updates) {
+        if (ColorProperties.indexOf(key) !== -1) {
+          updates[key] = processColor(updates[key]);
+        }
       }
-    }
 
-    viewDescriptors.value.forEach((viewDescriptor) => {
-      _updateProps(
-        viewDescriptor.tag,
-        viewDescriptor.name || 'RCTView',
-        updates
-      );
-    });
-  };
+      viewDescriptors.value.forEach((viewDescriptor) => {
+        _updatePropsFabric(viewDescriptor.shareableNode, updates);
```

Comment:
Could it be something like this instead of the whole function repetition?
```suggestion
  if (global._IS_FABRIC) {
        _updatePropsFabric(viewDescriptor.shareableNode, updates);
 } else {
   _updatePropsPaper(
          viewDescriptor.tag,
          viewDescriptor.name || 'RCTView',
          updates
        );
 }
```

Reply:
Yes, it could, but I suspect that it's better to have two different worklets and choose one of them during initialization than check architecture using `if` inside worklet on each props update.